### PR TITLE
Delete downloaded archive

### LINKED ISSUE #818: Delete Bitbucket Server archives downloaded by the CLI by default, but add an option to keep them
When running a migration from Bitbucket Server, the migration archive is kept in the `bbs_archive_downloads` directory.

Rather than leaving this trace on the user's machine - which could create security concerns - we should default to automatically deleting the file once it has been uploaded to blob storage (and indeed, when the CLI exits).

Whilst it should be deleted by default, we should still offer an option (e.g. `--keep-archive`) to keep a copy of the archive. This is useful for debugging purposes.

## Todo
- [x] Try to delete the archive after successful upload, in case of failure to delete log aa warning message.
- [x] Add `--keep-archive` option to both `bbs2gh migrate-repo` and `bbs2gh generate-script` commands.

### PR DESCRIPTION
**Pull request by ArinGhazarian**
Closes #818 

This PR deleted the downloaded archive after successfully uploading it. 

Note: I made a decision not to delete it in case of an upload failure but I can change it based on feedback. 

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)
- [x] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

**Review by synthead — APPROVED** [review]

File: src/Octoshift/FileSystemProvider.cs [hunk 1]
```diff
@@ -26,4 +26,12 @@ public virtual async ValueTask WriteAsync(FileStream fileStream, ReadOnlyMemory<
 
         await fileStream.WriteAsync(buffer, cancellationToken);
     }
+
+    public virtual void Delete(string path)
```
Comment (timrogers) [inline src/Octoshift/FileSystemProvider.cs:30]: What do you think of renaming this function so it's obvious what happens if the file isn't found?

```suggestion
    public virtual void DeleteIfExists(string path)
```

**Review by timrogers — COMMENTED** [review]
This is looking good! As discussed on Slack, I think we should still delete the file if the upload to blob storage fails.

File: src/bbs2gh/Commands/GenerateScriptCommand.cs [hunk 2]
```diff
@@ -96,6 +97,10 @@ public class GenerateScriptCommand : CommandBase<GenerateScriptCommandArgs, Gene
 
     public Option<bool> Verbose { get; } = new("--verbose");
 
+    public Option<bool> KeepArchive { get; } = new(
+        name: "--keep-archive",
+        description: "Keeps the downloaded export archive after successfully uploading it.");
```
[the remark this answers is not in the document]
Comment (ArinGhazarian) [reply src/bbs2gh/Commands/GenerateScriptCommand.cs]: Addressed in 0bf2055c9a3573c78aa0dd5fee98ef35b8fa4e8c

File: src/bbs2gh/Commands/MigrateRepoCommand.cs [hunk 3]
```diff
@@ -146,6 +147,10 @@ public class MigrateRepoCommand : CommandBase<MigrateRepoCommandArgs, MigrateRep
 
     public Option<bool> Verbose { get; } = new("--verbose");
 
+    public Option<bool> KeepArchive { get; } = new(
+        name: "--keep-archive",
+        description: "Keeps the downloaded export archive after successfully uploading it.");
```
[the remark this answers is not in the document]
Comment (ArinGhazarian) [reply src/bbs2gh/Commands/MigrateRepoCommand.cs]: addressed in 0bf2055c9a3573c78aa0dd5fee98ef35b8fa4e8c

File: src/Octoshift/FileSystemProvider.cs (hunk 1 above)
Comment (ArinGhazarian) [reply src/Octoshift/FileSystemProvider.cs]: addressed in 0bf2055c9a3573c78aa0dd5fee98ef35b8fa4e8c

**Review by timrogers — APPROVED** [review]

**Pull request merged**